test/e2e/healthcheck_test.go

```diff
@@ -0,0 +1,185 @@
+//go:build e2e
+
+package e2e
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/strings/slices"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kuadrant/dns-operator/api/v1alpha1"
+	"github.com/kuadrant/dns-operator/test/e2e/helpers"
+)
+
+// Test Cases covering multiple creation and deletion of health checks
+var _ = Describe("Health Check Test", Serial, func() {
+	Context("DNS Provider health checks", func() {
+		It("creates health checks for a health check spec", func(ctx SpecContext) {
+			probeHost := "app." + testZoneDomainName
+			healthChecksSupported := false
+			if slices.Contains(supportedHealthCheckProviders, strings.ToLower(testDNSProvider)) {
+				healthChecksSupported = true
+			}
+
+			provider, err := providerForManagedZone(ctx, testManagedZone)
+			Expect(err).To(BeNil())
+
+			By("creating a DNS Record")
+			dnsRecord := &v1alpha1.DNSRecord{}
+			err = helpers.ResourceFromFile("./fixtures/healthcheck_test/geo-dnsrecord-healthchecks.yaml", dnsRecord, helpers.GetTestEnv)
+			Expect(err).ToNot(HaveOccurred())
```

Comment:
You are creating this record but not ensuring it gets removed on failure. see https://github.com/Kuadrant/dns-operator/blob/main/test/e2e/single_record_test.go#L48-L52

Reply:
Another issue here is that we aren't creating a suite or test specific host to use see https://github.com/Kuadrant/dns-operator/blob/main/test/e2e/single_record_test.go#L36-L38

This is important as it allows the e2e tests to be executed with a set of unique hosts that won't conflict with each other. As this is we wouldn't be able to have the e2e test workflow execute on more than one PR at a time. Also if it fails to cleanup it will mess up the next run.

Reply:
I was wondering if I needed to use those name generators or not, cheers. I've updated it now.